### DISCUSSION TITLE
Different versions of trivy generating different reports for same docker image

### DISCUSSION THREAD

**Discussion by chandrash3khar** [aquasecurity/trivy#4859] [Q&A] [answered]
### Question

I am working on upgrading the trivy versions in our CI pipelines. We are upgrading from 0.16.0 to 0.43.1 and while doing the due diligence I have noticed while comparing the scan reports from both the versions that the reports was completely diifferent. I was expecting the newer version to detect more vulnerabilities on top what older version is able to. But I am seeing higher number of vulnerabilties on the older version while with the newer one number is quite low.

**New Version:**

```
dockerimage:tag
Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)
Node.js (node-pkg)
==================================
Total: 21 (UNKNOWN: 0, LOW: 1, MEDIUM: 16, HIGH: 2, CRITICAL: 2)

```
**Old Version:**

```
dockerimage:tag
Total: 257 (UNKNOWN: 0, LOW: 163, MEDIUM: 94, HIGH: 0, CRITICAL: 0)
package-lock.json
==================================
Total: 2 (UNKNOWN: 0, LOW: 0, MEDIUM: 2, HIGH: 0, CRITICAL: 0)
root/.npm/_npx/0d92fdd791cc1044/package-lock.json
=================================================
Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)
root/.npm/_npx/98ef08434c83cc95/package-lock.json
=================================================
Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)
```

Wanted to understand here if both the versions refer a different database or what is the cause of this difference.

### Target

Container Image

### Scanner

Vulnerability

### Output Format

Table

### Mode

Standalone

### Operating System

Ubuntu16

### Version

```bash
0.16.0
and
0.43.1
```

**Comment by knqyf263** [comment, marked as answer]
The newer version sess installed packages, while the older version just reads package-lock.json. It means the older version detects packages even not installed. It was improved.
